tests/boot/boot_to_desktop.pm

```diff
@@ -20,12 +20,12 @@ use version_utils qw(is_upgrade is_sles4sap);
 sub run {
     my ($self) = @_;
     $self->{in_boot_desktop} = 1;
-    # We have tests that boot from HDD and wait for DVD boot menu's timeout, so
-    # the timeout here must cover it. UEFI DVD adds some 60 seconds on top.
-    my $timeout = get_var('UEFI') ? 140 : 80;
-    # Add additional 60 seconds if the test suite is migration as reboot from
-    # pre-migration system may take an additional time.
-    $timeout += 60 if get_var('PATCH') || get_var('ONLINE_MIGRATION');
+    # In some cases default wait for boot should be increased due to different reasons.
+    # currently we awere about such reasons :
+    # - tests which boot from HDD and wait for boot menu's timeout ( UEFI DVD)
+    # - reboot from pre-migration system
+    # - skipping GRUB menu detection for unknown reason slowdown boot process
+    my $timeout = get_var('UEFI') || get_var('KEEP_GRUB_TIMEOUT') || get_var('PATCH') || get_var('ONLINE_MIGRATION') ? 140 : 80;
```

Comment:
Looks like change in the behavior, because previously in case UEFI and migration, we would get 200 seconds timeout for booting.
For the problem you are trying to solve, we actually should not call `wait_grub` and there is so called `ready_time` argument which is 300 seconds by default which is enough to boot the system. So I would suggest adjusting this condition, so we go further with the flow: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/opensusebasetest.pm#L616

Reply:
ah you mean case when `UEFI` **and** for example `PATCH` variables are defined ? yeah I haven't thought about such scenario ...

Reply:
> For the problem you are trying to solve, we actually should not call `wait_grub` and there is so called `ready_time` argument which is 300 seconds by default which is enough to boot the system. So I would suggest adjusting this condition, so we go further with the flow: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/opensusebasetest.pm#L616


exactly try to do this in #8142 , so closing this one